engine/graphics/src/null/graphics_null.cpp

```diff
@@ -1712,6 +1735,18 @@ namespace dmGraphics
         NullContext* context       = (NullContext*) _context;
         uint16_t param_array_index = (uint16_t) (size_t) data;
         SetTextureAsyncParams ap   = GetSetTextureAsyncParams(context->m_SetTextureAsyncState, param_array_index);
+
+        {
+            ASSET_CONTAINER_LOCK_BEGIN(context);
+            Texture* tex = GetAssetFromContainer<Texture>(context->m_AssetHandleContainer, ap.m_Texture);
+            if (tex)
+            {
+                SetTexture(ap.m_Texture, ap.m_Params);
+                tex->m_DataState &= ~(1<<ap.m_Params.m_MipMap);
+            }
+            ASSET_CONTAINER_LOCK_END(context);
```

Comment:
This is why we need the lock now, since we do the actual upload in the thread. So we need to protect the asset container so other threads don't start fiddling with its memory